src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -171,6 +171,15 @@ private async Task<IEnumerable<IAccount>> GetAccountsInternalAsync(ApiIds apiId,
         {
             Guid correlationId = Guid.NewGuid();
             RequestContext requestContext = CreateRequestContext(correlationId, cancellationToken);
+            
+            if (requestContext.Logger.IsLoggingEnabled(LogLevel.Info))
```

Comment:
So we usually check this before creating a logger message that has quite a bit of overhead. This doesn't seem to be too taxing so cant we let the logger adapter determine if this should be logged or not?

Reply:
This is in the hot path ... Any optimization counds. Besides at line 179-180 all those string interpolations are not free and will lead to some allocations etc.